project1/report.Rmd

```diff
@@ -59,16 +58,126 @@ data[, serum_cat := cut(serum_bilirubin, c(0, 1.1, 3.3, max(serum_bilirubin)))]
 
 # Analysis of the linear regression
 
+```{r, echo=TRUE, eval=TRUE}
+library(flexsurv)
+
+# Data input and cleaning
+
+# Reading in data
+data <- read.table("data.csv", 
+                   quote="\"", comment.char="")
+
+# Renaming variables
+colnames(data) <- c("caseid", "time", "status", "drug", "age", "sex", "ascites",
+                    "hepatomegaly", "spiders", "edema", "bilirubin", "cholesterol",
+                    "albumin", "urine_copper", "alk_phosphatase", "sgot", 
+                    "triglicerides", "platelets", "prothrombin", "hist_stage")
+
+# Combining censored cases
+data$dead <- ifelse(data$status == 0 | data$status == 1, 0, 1)
+table(data$dead)
+
+# Creating catergorical factor for bilirubin
+data$bili_cat <- ifelse(data$bilirubin < 1.1, "< 1.1", 
+                        ifelse(data$bilirubin >= 1.1 & data$bilirubin <= 3.3, "1.1-3.3", ">3.3"))
+data$bili_cat <- factor(data$bili_cat, levels = c("< 1.1", "1.1-3.3", ">3.3"))
+table( data$bili_cat)
+
+# Creating age in years
+data$ageinyear <- data$age/365.25
+```
+
 ## Excluding censored observations
 
+```{r, echo=TRUE, eval=TRUE}
+# univariate analysis for the treatment (drug)
+summary(lm(time ~ drug, data = data[data$dead==1,]))
+```
+
+```{r, echo=TRUE, eval=TRUE}
+# univariate analysis for age
+summary(lm(time ~ ageinyear, data = data[data$dead==1,]))
+```
+
+```{r, echo=TRUE, eval=TRUE}
+# univariate analysis for serum bilirubin
+summary(lm(time ~ bili_cat, data = data[data$dead==1,]))
+```
+
+```{r, echo=TRUE, eval=TRUE}
+# multivariate analysis for all three covariates
+summary(lm(time ~ drug + ageinyear + bili_cat, data = data[data$dead==1,]))
+```
+
 ## Treating censored times as death times
 
+```{r, echo=TRUE, eval=TRUE}
+# univariate analysis for the treatment (drug)
+summary(lm(time ~ drug, data = data))
+```
+
+```{r, echo=TRUE, eval=TRUE}
+# univariate analysis for age
+summary(lm(time ~ ageinyear, data = data))
+```
+
+```{r, echo=TRUE, eval=TRUE}
+# univariate analysis for serum bilirubin
+summary(lm(time ~ bili_cat, data = data))
+```
+
+```{r, echo=TRUE, eval=TRUE}
+# multivariate analysis for all three covariates
+summary(lm(time ~ drug + ageinyear + bili_cat, data = data))
+```
+

```

Comment:
Just another reminder that subheadings for each of these code blocks would be very helpful too